controllers/chat/ChatController.js

```diff
@@ -471,14 +471,40 @@ module.exports = {
       const findOrCreateChannel = await channel.create();
       findOrCreateChannel.members = await Promise.all(
         findOrCreateChannel.members.map(async (member) => {
-          if (member.role !== 'owner') return member;
+          if (member.role !== 'owner' && !member.user?.username?.includes('Anonymous'))
+            return member;
 
-          const anonInfo = await ChatAnonUserInfo.findOne({
+          let anonInfo = await ChatAnonUserInfo.findOne({
             where: {
               channel_id: findOrCreateChannel.channel.id,
               my_anon_user_id: member.user_id
             }
           });
+          if (!anonInfo && oldChannelId && member.user_id !== req.userId) {
+            let oldAnonInfo = await ChatAnonUserInfo.findOne({
+              where: {
+                channel_id: oldChannelId,
+                my_anon_user_id: {
+                  [Sequelize.Op.ne]: req.userId
+                }
+              }
+            });
+            if (oldAnonInfo) {
+              anonInfo = oldAnonInfo;
+              await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+                ChatAnonUserInfo,
+                findOrCreateChannel.channel.id,
+                member.user_id,
+                req?.userId,
+                {
+                  anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+                  anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+                  anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+                  anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name
+                }
+              );
+            }
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code has been significantly modified to handle anonymous users more effectively. It now checks if a member is not an owner and does not have 'Anonymous' in their username. If such a user is found, it attempts to find an existing `ChatAnonUserInfo` record using the channel ID and user ID. If no record is found and an `oldChannelId` is provided, it searches for an `oldAnonInfo` record using the `oldChannelId` and a different user ID. If this record exists, a new `anonInfo` record is created with the same properties as the `oldAnonInfo`.
```diff
-          if (member.role !== 'owner') return member;
- 
-          const anonInfo = await ChatAnonUserInfo.findOne({
-             where: {
-               channel_id: findOrCreateChannel.channel.id,
-               my_anon_user_id: member.user_id
-             }
-           });
+          if (member.role !== 'owner' && !member.user?.username?.includes('Anonymous'))
+            return member;
+  
+          let anonInfo = await ChatAnonUserInfo.findOne({
+             where: {
+               channel_id: findOrCreateChannel.channel.id,
+               my_anon_user_id: member.user_id
+             }
+           });
+          if (!anonInfo && oldChannelId && member.user_id !== req.userId) {
+            let oldAnonInfo = await ChatAnonUserInfo.findOne({
+              where: {
+                channel_id: oldChannelId,
+                my_anon_user_id: {
+                  [Sequelize.Op.ne]: req.userId
+                }
+              }
+            });
+            if (oldAnonInfo) {
+              anonInfo = oldAnonInfo;
+              await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+                ChatAnonUserInfo,
+                findOrCreateChannel.channel.id,
+                member.user_id,
+                req?.userId,
+                {
+                  anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+                  anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+                  anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+                  anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name
+                }
+              );
+            }
+          }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->